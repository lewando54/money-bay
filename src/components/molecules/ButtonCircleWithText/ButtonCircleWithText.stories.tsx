/* eslint-disable @typescript-eslint/consistent-type-assertions */
import { StoryObj, Meta } from '@storybook/react-native'

import ButtonCircleWithText from './ButtonCircleWithText'

const meta = {
	component: ButtonCircleWithText,
	argTypes: {
        // To write
	}
} satisfies Meta<typeof ButtonCircleWithText>

export default meta
type Story = StoryObj<typeof meta>

export const Default: Story = {
	args: {
        // To write
	}
}
