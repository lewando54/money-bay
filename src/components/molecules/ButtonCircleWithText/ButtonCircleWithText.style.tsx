import { Dimensions, StyleSheet } from 'react-native'
import { MAIN_DARK } from '../../../styling/GlobalTheme.style'

const { width } = Dimensions.get('screen')

export default StyleSheet.create({
    container: {
        justifyContent: 'center',
        width: width * 0.16,
        height: 95,
        elevation: 20,
        shadowColor: MAIN_DARK,
        shadowOffset: {width: 0, height: 4},
        shadowRadius: 16,
        shadowOpacity: 0.08,
        gap: 10
    },
    text: {
        color: MAIN_DARK,
        textAlign: 'center'
    }
})
