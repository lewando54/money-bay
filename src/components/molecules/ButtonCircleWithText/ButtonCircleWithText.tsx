import React from 'react'
import { TouchableOpacity, Text, Dimensions } from 'react-native'
import { AntDesign } from '@expo/vector-icons'
import buttonCircleWithTextStyle from './ButtonCircleWithText.style'
import CircleWithIcon from '../../atoms/CircleWithIcon/CircleWithIcon'
import GlobalThemeStyle from '../../../styling/GlobalTheme.style'

interface IButtonCircleWithTextProps {
    circleBackgroundColor?: string
    icon?: React.ReactNode
    text?: string
    onClick?: () => void
    testID?: string
}

export default function ButtonCircleWithText({
    circleBackgroundColor='#3EB290',
    icon=<AntDesign size={24} name={'creditcard'} color={'#fff'}/>,
    text='Top-Up Payment',
    onClick,
    testID='square-button-with-icon'
}: IButtonCircleWithTextProps){
    const { width, height } = Dimensions.get('screen')

    return (
        <TouchableOpacity onPress={onClick} testID={testID} style={buttonCircleWithTextStyle.container}>
            <CircleWithIcon icon={icon} backgroundColor={circleBackgroundColor} size={width * 0.16} castShadow={true}/>
            <Text style={[GlobalThemeStyle.text_SemiBold, buttonCircleWithTextStyle.text]}>{text}</Text>
        </TouchableOpacity>
    )
}
