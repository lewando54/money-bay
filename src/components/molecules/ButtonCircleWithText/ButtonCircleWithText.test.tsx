import React from 'react'
import renderer from 'react-test-renderer'
import { render, fireEvent } from '@testing-library/react-native'
import ButtonCircleWithText from './ButtonCircleWithText'
describe('ButtonCircleWithText', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<ButtonCircleWithText />)
    expect(tree).toMatchSnapshot()
  })

  it('should call onClick when pressed', () => {
    const onClick = jest.fn()
    const { getByTestId } = render(<ButtonCircleWithText onClick={onClick} testID='test'/>)
    fireEvent.press(getByTestId('test'))
    expect(onClick).toBeCalled()
  })
})
