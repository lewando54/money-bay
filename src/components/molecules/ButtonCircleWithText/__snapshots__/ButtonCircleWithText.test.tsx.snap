// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonCircleWithText should render correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "elevation": 20,
      "gap": 10,
      "height": 95,
      "justifyContent": "center",
      "opacity": 1,
      "shadowColor": "#1B1D4D",
      "shadowOffset": {
        "height": 4,
        "width": 0,
      },
      "shadowOpacity": 0.08,
      "shadowRadius": 16,
      "width": 120,
    }
  }
  testID="square-button-with-icon"
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#3EB290",
          "borderRadius": 240,
          "height": 120,
          "justifyContent": "center",
          "width": 120,
        },
        {
          "elevation": 20,
          "shadowColor": "#16196670",
          "shadowOffset": {
            "height": 6,
            "width": 0,
          },
          "shadowOpacity": 0.08,
          "shadowRadius": 16,
        },
      ]
    }
  >
    <Text />
  </View>
  <Text
    style={
      [
        {
          "color": "white",
          "fontFamily": "MulishSemiBold",
        },
        {
          "color": "#1B1D4D",
          "textAlign": "center",
        },
      ]
    }
  >
    Top-Up Payment
  </Text>
</View>
`;
